aws/resource_aws_transfer_user_test.go

```diff
@@ -650,3 +659,122 @@ resource "aws_transfer_user" "foo" {
 }
 `, rName, rName)
 }
+
+func testAccAWSTransferUserConfig_homeDirectoryMappings(rName string) string {
+	return testAccAWSTransferUserConfig_base + fmt.Sprintf(`
+
+resource "aws_iam_role" "foo" {
+  name = "tf-test-transfer-user-iam-role-%s"
+
+  assume_role_policy = <<EOF
+{
+	"Version": "2012-10-17",
+	"Statement": [
+		{
+			"Effect": "Allow",
+			"Principal": {
+				"Service": "transfer.amazonaws.com"
+			},
+			"Action": "sts:AssumeRole"
+		}
+	]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "foo" {
+  name = "tf-test-transfer-user-iam-policy-%s"
+  role = "${aws_iam_role.foo.id}"
```

Comment:
Nit: We are now using Terraform 0.12+ syntax in test configurations, will adjust on merge. 👍